### PR TITLE
feat: add erc7484 to msaadvanced

### PR DESCRIPTION
**Pull request by kopy-kat**

### CODE REVIEW

File: src/core/RegistryAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    modifier withRegistry(address module, uint256 moduleTypeId) {
+        IERC7484 registry = $registry;
+        if (address(registry) != address(0)) {
+            registry.check(module, moduleTypeId);
+        }
+        _;
+    }
+
+    function setRegistry(
```
Comment (livingrockrises) [inline src/core/RegistryAdapter.sol:19]: Where is the latest code to the registry? (repo link)
Is it audited?
do you guys have deployed registry on some networks?

File: src/core/RegistryAdapter.sol [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        IERC7484 registry,
+        address[] calldata attesters,
+        uint8 threshold
+    )
+        external
```
Comment (livingrockrises) [inline src/core/RegistryAdapter.sol:24]: What should be the access controls on this method?

**Comment by livingrockrises** [comment]
alternate way would also be to have this .check happen inside a hook

File: src/core/RegistryAdapter.sol (hunk 1 above)
Comment (kopy-kat) [reply src/core/RegistryAdapter.sol:19]: https://github.com/rhinestonewtf/registry/

Audit is done but in remediation atm

File: src/core/RegistryAdapter.sol (hunk 2 above)
Comment (kopy-kat) [reply src/core/RegistryAdapter.sol:24]: good point, ep or self would make sense imo

**Comment by kopy-kat** [comment]
yep, we have a hook example here: https://github.com/rhinestonewtf/modulekit/blob/examples/audit-ready/examples/src/RegistryHook/RegistryHook.sol

**Comment by kopy-kat** [comment]
one thing that I thought could be useful is demonstrate more advanced integrations in msa advanced to highlight some additional features that are outside of 7579 but accounts could use (eg hooks, 7484, policy/signers) - wdyt

File: src/core/RegistryAdapter.sol (hunk 2 above)
Comment (livingrockrises) [reply src/core/RegistryAdapter.sol:24]: yup

**Comment by livingrockrises** [comment]
> one thing that I thought could be useful is demonstrate more advanced integrations in msa advanced to highlight some additional features that are outside of 7579 but accounts could use (eg hooks, 7484, policy/signers) - wdyt

yes I do want to add tests in 7579 repo (reference / biconomy v3 -calling this 'Nexus' now) using module-kit
using some real 'useful executor' and real hooks which cover
- installation and 7484
- executor limits
etc
can chat more

File: src/core/RegistryAdapter.sol (hunk 1 above)
Comment (livingrockrises) [reply src/core/RegistryAdapter.sol:20]: also probably good to rename it to setModuleRegistry and have ERC7484 mention in the natspecs

File: src/MSAAdvanced.sol [hunk 3]
```diff
@@ -87,6 +88,7 @@ contract MSAAdvanced is IMSA, ExecutionHelper, ModuleManager, HookManager {
         external
         payable
         onlyExecutorModule
+        withRegistry(msg.sender, MODULE_TYPE_EXECUTOR)
```
Comment (livingrockrises) [inline src/MSAAdvanced.sol:91]: this may not be needed as executor attestation would have been checked upon installation.

Comment (kopy-kat) [reply src/MSAAdvanced.sol:91]: this is more for ongoing security, eg an executor is installed and later a bug is found and the attestation revoked

**Pull request merged**